src/Compilers/VisualBasic/Portable/CommandLine/VisualBasicCommandLineParser.vb

```diff
@@ -446,7 +447,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
 #If DEBUG Then
                     Case "attachdebugger"
-                        Debugger.Launch()
+                        If RuntimeInformation.IsOSPlatform(OSPlatform.Windows) Then
+                            ' Debugger.Launch() only works on Windows
+                            Debugger.Launch()
+                        Else
+                            Dim timeout = TimeSpan.FromMinutes(2)
+#If NET Then
```

Comment:
Do we still need conditional directives here given that the implementation of `Environment.ProcessId` already has them?

Reply:
This is VB.